### PR TITLE
feat: enhance i18next typing

### LINKED ISSUE #89: 🤔 [ISSUE] - missing ui translations
### Description

i18n translation files are missing keys. 

**en/translation.json file is missing those translation keys :** 
- button.toggle
- label.inputs
- label.outputs
- message.button_block_message_label
- placeholder.mark_as_default

**fr/translation.json file is missing those translation keys :** 
- label.button_block_message_label
- label.inputs
- label.outputs
- placeholder.type
- placeholder.unselect
- button.mark_as_default
- button.toggle

### Reproduction steps

```bash
1. Go to frontend/src/i18n/rn/translation.json and/or frontend/src/i18n/fr/translation.json
2. Check the missed keys mentioned in the description
```

### Logs

_No response_

### Browsers

Chrome

### OS

Linux

### PR DESCRIPTION
**Pull request by yassinedorbozgithub**
# Motivation
The main motivation is to enhance i18next typing and to fix i18next translation files missing keys.

Fixes #89

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/components/settings/SettingInput.tsx [hunk 1]
```diff
@@ -35,10 +36,10 @@ const SettingInput: React.FC<RenderSettingInputProps> = ({
   isDisabled = () => false,
 }) => {
   const { t } = useTranslation();
-  const label = t(`label.${setting.label}`, {
+  const label = t(getNestedTranslation("label", setting.label), {
```
Comment (marrouchi) [inline frontend/src/components/settings/SettingInput.tsx:39]: I suggest we create a hook called useTranslate() in which we call t(getNestedTranslation(....))

```
const t = useTranslate() // and then use t("label", setting.label)
```

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: frontend/src/app-components/tables/columns/getColumns.tsx [hunk 2]
```diff
@@ -27,22 +27,35 @@ import {
...
+  Values = "Values",
+  Manage_Roles = "Manage_Roles",
+  Permissions = "Permissions",
+  Content = "Content",
+  Fields = "Fields",
+  Manage_Labels = "Manage_Labels",
+  Toggle = "Toggle",
 }
 
+const actionColumnLabelMap: Record<ActionColumnLabel, TTranslationKeys> = {
```
[the remark this answers is not in the document]
Comment (yassinedorbozgithub) [reply frontend/src/app-components/tables/columns/getColumns.tsx]: Change applied :white_check_mark:

File: frontend/src/components/settings/SettingInput.tsx (hunk 1 above)
Comment (yassinedorbozgithub) [reply frontend/src/components/settings/SettingInput.tsx]: Change applied
I'm thinking about the case when we use useTranslation and useTranslate hooks in the same file so we will get two returned function named t

File: frontend/src/types/common/object.types.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright © 2024 Hexastack. All rights reserved.
+ *
+ * Licensed under the GNU Affero General Public License v3.0 (AGPLv3) with the following additional terms:
+ * 1. The name "Hexabot" is a trademark of Hexastack. You may not use this name in derivative works without express written permission.
+ * 2. All derivative works must include clear attribution to the original creator and software, Hexastack and Hexabot, in a prominent location (e.g., in the software's "About" section, documentation, and README file).
+ * 3. SaaS Restriction: This software, or any derivative of it, may not be used to offer a competing product or service (SaaS) without prior written consent from Hexastack. Offering the software as a service or using it in a commercial cloud environment without express permission is strictly prohibited.
```
Comment (marrouchi) [inline frontend/src/types/common/object.types.ts:7]:
```suggestion
```

File: frontend/src/i18n/index.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import en from "@/i18n/en/translation.json";
```
Comment (marrouchi) [inline frontend/src/i18n/index.ts:1]: Let's add the copyright header

File: frontend/src/i18n/i18n.types.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { TOptionsBase } from "i18next";
```
Comment (marrouchi) [inline frontend/src/i18n/i18n.types.ts:1]: Let's add the copyright header

**Review by marrouchi — CHANGES_REQUESTED** [review]
Awesome work ! 👏 🥳  Left comments for minor changes

**Review by marrouchi — APPROVED** [review]

**Pull request merged**